src/sql/workbench/services/tableDesigner/browser/tableDesignerService.ts

```diff
@@ -11,16 +12,38 @@ import { TableDesignerInput } from 'sql/workbench/browser/editor/tableDesigner/t
 import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
 import { IAdsTelemetryService, ITelemetryEventProperties } from 'sql/platform/telemetry/common/telemetry';
 import { TelemetryAction, TelemetryView } from 'sql/platform/telemetry/common/telemetryKeys';
+import { IDialogService } from 'vs/platform/dialogs/common/dialogs';
+import { ILifecycleService } from 'vs/workbench/services/lifecycle/common/lifecycle';
 
 export class TableDesignerService implements ITableDesignerService {
 
 	constructor(@IEditorService private _editorService: IEditorService,
 		@IInstantiationService private _instantiationService: IInstantiationService,
-		@IAdsTelemetryService private _adsTelemetryService: IAdsTelemetryService) { }
+		@IAdsTelemetryService private _adsTelemetryService: IAdsTelemetryService,
+		@ILifecycleService private _lifecycleService: ILifecycleService,
+		@IDialogService private _dialogService: IDialogService) {
+		this._lifecycleService.onBeforeShutdown(async (event) => {
+			event.veto(this.confirmBeforeExit(), 'veto.tableDesigner');
+		});
+	}
 
 	public _serviceBrand: undefined;
 	private _providers = new Map<string, TableDesignerProvider>();
 
+	private async confirmBeforeExit(): Promise<boolean> {
+		let openTableDesignerEditors = this._editorService.editors.filter(e => e instanceof TableDesignerInput);
+		let dirtyEditors = openTableDesignerEditors.find(e => e.isDirty());
+		if (dirtyEditors) {
```

Comment:
I am afraid this won't work, dirtyEditors will be an empty array if nothing is dirty, you will still show the dialog.

Reply:
should check for `dirtyEditors.length !== 0`